### PR TITLE
Add ETH-related Merkle trie wrapper

### PR DESCRIPTION
**Pull request by kfichter**
## Description
Hello! This PR introduces a new contract `EthMerkleTrie.sol` that extends `MerkleTrie.sol`. The new contract includes various utility wrappers around `MerkleTrie.sol` for ETH-specific operations.

Builds upon #160 

## Questions
- Does the structure of `proveAccountState` and `updateAccountState` seem too verbose?

### Fixes
- Fixes [YAS-418](https://optimists.atlassian.net/browse/YAS-418)

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: packages/rollup-contracts/package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "fix": "prettier --config ../../prettier-config.json --write \"index.ts\" \"{deploy,test}/**/*.ts\"",
     "build": "waffle waffle-config.json && tsc -p .",
     "clean": "rimraf build/",
-    "deploy:rollup-chain": "yarn build && node ./build/deploy/rollup-chain.js"
+    "deploy:rollup-chain": "yarn build && node ./build/deploy/rollup-chain.js",
+    "womp": "mocha --require ts-node/register 'test/merklization/EthMerkleTrie.spec.ts' --timeout 20000"
```
Comment (karlfloersch) [inline packages/rollup-contracts/package.json:17]: Lol is `womp` a debugging thing that you forgot to remove?

Comment (kfichter) [reply packages/rollup-contracts/package.json:17]: Hahaha yes, whoops

Comment (kfichter) [reply packages/rollup-contracts/package.json]: Fixed

**Comment by kfichter** [comment]
Just merged master, if CI passes then I'll merge this!

**Pull request merged**